File: src/main/java/pl/training/workshop/m6/s12_onemany/step3/CancellationDesk.java
package pl.training.workshop.m6.s12_onemany.step3;

import java.time.LocalDateTime;
import java.util.List;

import pl.training.workshop.m6.s12_onemany.TicketData;
import pl.training.workshop.shared.Money;

/** Krok 3: rozróżnienie zniknęło - grupa jednego biletu zachowuje się jak bilet. */
public final class CancellationDesk {
    private final RefundService service = new RefundService();

    public Money refund(List<TicketData> tickets, LocalDateTime now) {
        return service.refund(TicketGroup.of(tickets), now);
    }
}

File: typescript/src/workshop/m6/s12_onemany/step3/CancellationDesk.ts
import type { Money } from '../../../shared/Money.js';
import type { LocalDateTime } from '../../../shared/time.js';
import type { TicketData } from '../TicketData.js';
import { RefundService } from './RefundService.js';
import { TicketGroup } from './TicketGroup.js';

/** Krok 3: rozróżnienie zniknęło - grupa jednego biletu zachowuje się jak bilet. */
export class CancellationDesk {
  private readonly service = new RefundService();

  refund(tickets: readonly TicketData[], now: LocalDateTime): Money {
    return this.service.refund(TicketGroup.of(tickets), now);
  }
}
